proxy/src/http/conn_pool.rs

```diff
@@ -0,0 +1,277 @@
+use pq_proto::StartupMessageParams;
+use std::fmt;
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::Mutex;
```

Comment:
We can also use std mutex here (or even parking_lot), because we don't `.await` anything else while holding the lock. Not sure which mutex is faster though.

Reply:
Tokio mutex internally uses a std (or parking_lot) mutex. It is necessarily slower.

Std and parking_lot mutexes are equivalent on linux since around 1.65 AFAIU.

Reply:
Switched to parking_lot